Gems/Atom/RHI/Metal/Code/Source/Platform/Mac/RHI/Metal_RHI_Mac.cpp

```diff
@@ -43,11 +43,38 @@ namespace Platform
         }
         return physicalDeviceList;
     }
+    
+    float GetRefreshRate()
+    {
+        CGDirectDisplayID display = CGMainDisplayID();
+        CGDisplayModeRef currentMode = CGDisplayCopyDisplayMode(display);
+        return CGDisplayModeGetRefreshRate(currentMode);
+    }
 
-    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval)
+    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval, float refreshRate)
     {
-        AZ_UNUSED(syncInterval);
-        [mtlCommandBuffer presentDrawable:drawable];
+        bool hasPresentAfterMinimumDurationAPI = false;
+        bool framePresented = false;
+        float presentAfterMinimumDuration = syncInterval / refreshRate;
```

Comment:
Nit: Needs a comment on what this is calculating.
it's really a short form for seconds per frame (1/refreshrate) * num frames (sync interval).